gym_minigrid/envs/dynamic_minigrid.py

```diff
@@ -39,18 +39,49 @@ def _gen_grid(self, width, height):
 
     def alter(self, prob_dict, visibility_check=True):
         """
-        Changes a single element of the environment.
+        Changes a single element of the environment. Automatically checks whether the environment
+        can be (empirically in 5000 random steps) solved.
+        Applies another change of the same alteration if environment is not solvable.
 
         :param prob_dict: dict. Dictionary of probabilties for each type of altering
             (change start or goal position, add wall/lava). Elements must sum to 1
         :param visibility_check: bool. If true, checks whether the agent can see the reward
             at the start and rejects such a solution.
-        :return: boolean. True if the environment can be solved empirically within 10'000 steps.
+        :return: None
         """
 
         if sum(prob_dict.values()) != 1.0:
             raise ValueError('Probabilities do not sum to 1')
 
+        def is_solvable():
+            # empirical check: let a random agent take max_steps and see if it visited the goal
+            max_steps = 5000
+
+            if self.height * self.width > 100:
+                warnings.warn(f"Solvability takes {max_steps} with a random agent, "
+                              f"thus might wrong in large grids ", UserWarning)
+
+            reachable_pos = [self.agent_start_pos]
+            for _ in range(max_steps):
+                # take a (random) step
+                action = self.np_random.randint(low=0, high=3)  # 0 turn left, 1 turn right, 2 move
+                self.step(action)
+                reachable_pos.append(tuple(self.agent_pos))
+                if self.grid.get(*self.agent_pos) is not None and \
+                        self.grid.get(*self.agent_pos).type == Lava().type:
```

Comment:
```suggestion
                        isinstance(self.grid.get(*self.agent_pos), Lava):
```
i think this is how we should compare types?